### DISCUSSION TITLE
Data flow hierarchy between files

### DISCUSSION THREAD

**Discussion by radiorambo** [sveltejs/kit#12176] [General]
can some one explain data hierarchy in +page.js, +page.server.js, +page.svelte, +layout.svelte, +layout.js, +layout.svelte, hooks.server.js, hooks,client.js, hooks.js, +server.js.

If I have a +page.svelte page which needs data defined in each of the above files, how will the data flow look like. Which files run one after another?

**Comment by alyshareinard** [comment]
This page has a great description: https://joyofcode.xyz/sveltekit-data-flow
